sources.go

```diff
@@ -101,33 +106,72 @@ func adjust(img image.Image, mode string, scale float64) image.Image {
 
 	debug("Adjusting image")
 
-	re_width := 1404
-	re_height := 1872
+	reWidth := 1404
+	reHeight := 1872
 
 	if mode == "fill" {
 		// scale image to remarkable width
 		// imaging resize is slow for some reason, use other library
 		// img = imaging.Resize(img, re_width, 0, imaging.Linear)
-		img = resize.Resize(uint(re_width), 0, img, resize.Bilinear)
+		img = resize.Resize(uint(reWidth), 0, img, resize.Bilinear)
 		// cut off parts of image that overflow
-		img = imaging.Crop(img, image.Rect(0, 0, re_width, re_height))
+		img = imaging.Crop(img, image.Rect(0, 0, reWidth, reHeight))
 
 	} else if mode == "center" {
 	} else {
 		debug("Invalid mode")
 	}
 	if scale != 1 {
-		img_width := float64(img.Bounds().Max.X)
-		img = resize.Resize(uint(scale * img_width), 0, img, resize.Bilinear)
+		imgWidth := float64(img.Bounds().Max.X)
+		img = resize.Resize(uint(scale*imgWidth), 0, img, resize.Bilinear)
 	}
 
 	// put image in center of screen
 	background := imaging.New(
-		re_width,
-		re_height,
+		reWidth,
+		reHeight,
 		color.RGBA{255, 255, 255, 255},
 	)
 	img = imaging.PasteCenter(background, img)
 
 	return img
 }
+
+func addText(img image.Image, y int, label string) image.Image {
+
+	debug("Adding text to image: ", label)
+
+	ttfData, err := ioutil.ReadFile("./NotoSerif-Regular.ttf")
+	if err != nil {
+		ttfData, err = ioutil.ReadFile("/usr/share/fonts/ttf/noto/NotoSerif-Regular.ttf")
+	}
+	check(err, "Couldn't load TTF font")
+
+	ttf, err := truetype.Parse(ttfData)
+	check(err, "Couldn't parse font data")
+
+	face := truetype.NewFace(ttf, &truetype.Options{
+		Size: 24,
+		DPI:  72,
+	})
+	check(err, "Couldn't create font face")
+
+	textColor := color.RGBA{50, 50, 50, 255}
+
+	d := &font.Drawer{
+		Dst:  img.(*image.NRGBA),
+		Src:  image.NewUniform(textColor),
+		Face: face,
+	}
+
+	width := d.MeasureString(label)
+	x := (fixed.I(1404) - width) / 2.0
+
+	d.Dot = fixed.Point26_6{
+		X: x,
+		Y: fixed.I(y),
+	}
+	d.DrawString(label)
```

Comment:
Use the returned string from `get_xpath(url, subtitle, "html)` as the subtitle text instead of the user provided string directly.

Reply:
Bump